linalg/amgxsolver.hpp

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+//Reference:
+//Pi-Yueh Chuang, & Lorena A. Barba (2017).
+//AmgXWrapper: An interface between PETSc and the NVIDIA AmgX library. J. Open Source Software, 2(16):280, doi:10.21105/joss.00280
+
+#ifndef MFEM_AMGX_SOLVER
+#define MFEM_AMGX_SOLVER
+
+#include "../config/config.hpp"
+
+#ifdef MFEM_USE_AMGX
+
+#include <amgx_c.h>
+#ifdef MFEM_USE_MPI
+#include <mpi.h>
+#include "hypre.hpp"
+#else
+#include "operator.hpp"
+#include "sparsemat.hpp"
+#endif
+
+namespace mfem
+{
+
+class AmgXSolver : public Solver
+{
+
+public:
+
+   enum AMGX_MODE {SOLVER, PRECONDITIONER};
```

Comment:
Here (or maybe before the `DefaultParameters` method you should document what these defaults are, at least the basic pionts (ie, which smoother?)